### PR TITLE
Minor changes to openstack-nova and openstack-keystone

### PR DESCRIPTION
**Pull request by andreaturli**
- change some access modifiers to simplify the downstream provider impl
- modify the BasicAuthenticator in Keystone

@zack-shoylev, @jdaggett as openstack experts, could you give it a quick look please?

Thanks!

### CODE REVIEW

**Comment by andreaturli** [comment]
@nacx those are basic changes, do you think we can merge them for the upcoming release (1.9.1) 
It would be a huge help!

File: apis/openstack-nova/src/main/java/org/jclouds/openstack/nova/v2_0/compute/NovaComputeServiceAdapter.java [hunk 1]
```diff
@@ -127,16 +130,26 @@ public NovaComputeServiceAdapter(NovaApi novaApi, @Region Supplier<Set<String>>
...
-      logger.trace("<< server(%s)", server.getId());
+      final ServerCreated lightweightServer = novaApi.getServerApi(regionId).create(name, imageId, flavorId, options);
+      if (!retry(new Predicate<String>() {
+         @Override
+         public boolean apply(String serverId) {
+            Server server = novaApi.getServerApi(regionId).get(serverId);
+            return server != null && server.getAddresses() != null && !server.getAddresses().isEmpty();
+         }
+      }, 30 * 60, 1, SECONDS).apply(lightweightServer.getId())) {
+         final String message = format("Server %s was not created within %sms so it will be destroyed.", name, "30 * 60");
```
Comment (nacx) [inline apis/openstack-nova/src/main/java/org/jclouds/openstack/nova/v2_0/compute/NovaComputeServiceAdapter.java]: Where is the server destroyed?

Comment (andreaturli) [inline apis/openstack-nova/src/main/java/org/jclouds/openstack/nova/v2_0/compute/NovaComputeServiceAdapter.java]: ops, you're right! adding it now

**Comment by nacx** [comment]
lgtm. Thanks @andreaturli!

**Comment by andreaturli** [comment]
Merged at [master](http://git-wip-us.apache.org/repos/asf/jclouds/commit/aa64e0d3) and backported to [1.9.x](http://git-wip-us.apache.org/repos/asf/jclouds/commit/4ac9a73d)

**Pull request closed without merging**